Buddies.Tests/User/LoginTests.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using Buddies.API.IO;
+using Xunit;
+
+namespace Buddies.Tests.User;
+
+public class LoginTests : IClassFixture<TestWebApplicationFactory<Program>>
+{
+
+
+    private readonly HttpClient _client;
+    private readonly TestWebApplicationFactory<Program> _factory;
+
+    public LoginTests(TestWebApplicationFactory<Program> factory)
+    {
+        _factory = factory;
+        _client = _factory.CreateClient();
+    }
+
+    [Fact]
+    public async Task TestEmptyRequest()
+    {
+        var request = new LoginRequest
+        {
+            Email = "",
+            Password = ""
+        };
+
+        var response = await _client.PostAsJsonAsync("/api/v1/users/login", request);
+        Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+
+    }
+
+    [Fact]
+    public async Task TestUnauthorizedLogin()
+    {
+        var request = new RegisterRequest
+        {
+            FirstName = "John",
+            LastName = "Doe",
+            Email = "123@email.com",
+            Password = "Abc123."
+        };
+
+        var response = await _client.PostAsJsonAsync("/api/v1/users/register", request);
+        Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+        var loginRequest = new LoginRequest
+        {
+            Email = "123@email.com",
+            Password = "WRONGPASSWORD"
+        };
+
+        var loginResponse = await _client.PostAsJsonAsync("/api/v1/users/login", loginRequest);
+        Assert.Equal(HttpStatusCode.Unauthorized, loginResponse.StatusCode);
+    }
+
+
+    [Fact]
+    public async Task TestAuthorizedLogin()
+    {
+        var request = new RegisterRequest
+        {
+            FirstName = "John",
+            LastName = "Doe",
+            Email = "1234@email.com",
+            Password = "Abc123."
+        };
+
+        var response = await _client.PostAsJsonAsync("/api/v1/users/register", request);
+        Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+        var loginRequest = new LoginRequest
+        {
+            Email = "1234@email.com",
+            Password = "Abc123."
+        };
+        var loginResponse = await _client.PostAsJsonAsync("/api/v1/users/login", loginRequest);
+        Assert.Equal(HttpStatusCode.OK, loginResponse.StatusCode);
+    }
+
+    [Fact]
+    public async Task TestRefreshToken()
+    {
+        var request = new RegisterRequest
+        {
+            FirstName = "John",
+            LastName = "Doe",
+            Email = "12345@email.com",
+            Password = "Abc123."
+        };
```

Comment:
I would pull out user creation into a `createValidUser` function or something. Future tests will require creating users for projects and such and this would reduce the amount of code, and perhaps make some changes simpler. 

I would also think about some other things you could pull out into a test helpers file.

Reply:
Per not mixing different tasks inside 1 PR, we'll perform test refactoring in another PR (tracked in #16).